src/persist/src/indexed/mod.rs

```diff
@@ -367,23 +451,99 @@ impl<L: Log, B: Blob> Indexed<L, B> {
         Ok(())
     }
 
-    /// Creates, if necessary, a new unsealed and trace with the given external
-    /// stream name, returning the corresponding internal stream id.
-    ///
-    /// This method is idempotent: ids may be registered multiple times.
-    pub fn register(
+    // A test helper for Cmd::Register.
+    #[cfg(test)]
+    fn register(
         &mut self,
-        id_str: &str,
+        name: &str,
         key_codec_name: &str,
         val_codec_name: &str,
         res: PFutureHandle<Id>,
-    ) {
-        res.fill((|| {
-            self.drain_pending()?;
-            self.apply_unbatched_cmd(|state, _, _| {
-                state.do_register(id_str, key_codec_name, val_codec_name)
-            })
-        })());
+    ) -> bool {
+        self.apply(Cmd::Register(
+            name.into(),
+            (key_codec_name.to_owned(), val_codec_name.to_owned()),
+            res,
+        ))
+    }
+
+    /// Applies the given input to the state machine.
```

Comment:
nit: could we still have internal methods for the individual `Cmd`s. As it is, the method is a bit unwieldy. It might be that you wanted to ensure that no one can by accident think they should call these methods instead of going through `apply`, if so, then please ignore.

Reply:
threw something on your calendar for monday

Reply:
discussed this offline and decided this is okay as-is